lib/easypost.rb

```diff
@@ -1,157 +1,139 @@
-require 'cgi'
-require 'set'
-require 'openssl'
-require 'rest_client'
-require 'multi_json'
+require "base64"
+require "cgi"
+require "net/http"
 
 # Resources
-require 'easypost/version'
-require 'easypost/util'
-require 'easypost/object'
-require 'easypost/resource'
-require 'easypost/address'
-require 'easypost/parcel'
-require 'easypost/customs_item'
-require 'easypost/customs_info'
-require 'easypost/shipment'
-require 'easypost/rate'
-require 'easypost/postage_label'
-require 'easypost/scan_form'
-require 'easypost/refund'
-require 'easypost/insurance'
-require 'easypost/event'
-require 'easypost/batch'
-require 'easypost/tracker'
-require 'easypost/item'
-require 'easypost/order'
-require 'easypost/pickup'
-require 'easypost/pickup_rate'
-require 'easypost/printer'
-require 'easypost/print_job'
-require 'easypost/carrier_account'
-require 'easypost/user'
-require 'easypost/report'
-require 'easypost/webhook'
-
-require 'easypost/error'
+require "easypost/version"
+require "easypost/util"
+require "easypost/object"
+require "easypost/resource"
+require "easypost/address"
+require "easypost/parcel"
+require "easypost/customs_item"
+require "easypost/customs_info"
+require "easypost/shipment"
+require "easypost/rate"
+require "easypost/postage_label"
+require "easypost/scan_form"
+require "easypost/refund"
+require "easypost/insurance"
+require "easypost/event"
+require "easypost/batch"
+require "easypost/tracker"
+require "easypost/item"
+require "easypost/order"
+require "easypost/pickup"
+require "easypost/pickup_rate"
+require "easypost/printer"
+require "easypost/print_job"
+require "easypost/carrier_account"
+require "easypost/user"
+require "easypost/report"
+require "easypost/webhook"
+
+require "easypost/error"
 
 module EasyPost
-  @@api_key = nil
-  @@api_base = 'https://api.easypost.com/v2'
-  @@api_version = nil
-  @@open_timeout = 30
-  @@timeout = 60
-
-  def self.api_url(url='')
-    @@api_base + url
-  end
+  @api_key = nil
+  @api_base = "https://api.easypost.com"
 
   def self.api_key=(api_key)
-    @@api_key = api_key
+    @api_key = api_key
   end
 
   def self.api_key
-    @@api_key
+    @api_key
   end
 
   def self.api_base=(api_base)
-    @@api_base = api_base
+    @api_base = api_base
   end
 
   def self.api_base
-    @@api_base
-  end
-
-  def self.api_version=(version)
-    @@api_version = version
+    @api_base
   end
 
-  def self.api_version
-    @@api_version
-  end
-
-  def self.http_config
-    @@http_config ||= {
+  def self.reset_http_config
+    @http_config = {
       timeout: 60,
       open_timeout: 30,
       verify_ssl: OpenSSL::SSL::VERIFY_PEER,
       ssl_version: :TLSv1_2,
```

Comment:
note that this is deprecated and on modern Rubies, we should be setting min_version=:TLSv1_2 on the SSL Context object and leaving max_version empty. This will probably bite us when we roll out TLSv1.3 later this year, so it's probably worth fixing now.